src/Security/Authentication/test/ClaimActionTests.cs

```diff
@@ -41,6 +41,38 @@ public void CanMapArrayValueUserDataToClaims()
         Assert.Equal("role2", roleClaims[1].Value);
     }
 
+    [Fact]
+    public void CanMapSingleSubValueUserDataToClaim()
+    {
+        var userData = JsonDocument.Parse("{ \"name\": { \"subkey\": \"test\" } }");
```

Comment:
You've removed an IsNullOrEmpty check. Add a test case for `"{ \"name\": { \"subkey\": \"\" } }"` to see if you get the same behavior before and after?

Reply:
The value cannot be null, otherwise the last `TryGetProperty` would return `false`. If it was an empty string, than that is dealt with by the `base.AddClaim` which does have the check. I see this was merged, do you want me to add more tests? Sorry I wasn't available earlier.

Reply:
Thanks for explaining, no need for further work here.